README.md
# Create a project using react and CRA

Using CRA is optional

### npx create-react-app ui-consumer-app

# Install the ui library package into your project

### `npm i cc-ui-components`

# Start your project

### `npm start`

# Modify your public/index.html file to import the Source-Sans Pro font

```
<head>
...
<link rel="preconnect" href="https://fonts.gstatic.com">
<link href="https://fonts.googleapis.com/css2?family=Source+Sans+Pro:wght@400;600;700;900&display=swap" rel="stylesheet">
<style>
    body {
        font-family: 'Source Sans Pro', sans-serif;
    }
</style>
...
</head>
```

# Import the Source-Sans Pro font from the NPM package

If you don't want to rely on a CDN to import your font, you can alternatively include the font into your project  importing it from the NPM package in woff and woff2 formats from `cc-ui-components/dist/fonts` in a manner that suites your project the best.

```
source-sans-pro-v14-latin-italic.woff';
source-sans-pro-v14-latin-regular.woff';
source-sans-pro-v14-latin-200.woff';
source-sans-pro-v14-latin-200italic.woff';
source-sans-pro-v14-latin-200.woff';
source-sans-pro-v14-latin-200italic.woff';
source-sans-pro-v14-latin-200.woff';
source-sans-pro-v14-latin-200italic.woff';
source-sans-pro-v14-latin-200.woff';
source-sans-pro-v14-latin-200italic.woff';
source-sans-pro-v14-latin-200.woff';
source-sans-pro-v14-latin-200italic.woff';
source-sans-pro-v14-latin-italic.woff2';
source-sans-pro-v14-latin-regular.woff2';
source-sans-pro-v14-latin-200.woff2';
source-sans-pro-v14-latin-200italic.woff2';
source-sans-pro-v14-latin-200.woff';
source-sans-pro-v14-latin-200italic.woff2';
source-sans-pro-v14-latin-200.woff';
source-sans-pro-v14-latin-200italic.woff2';
source-sans-pro-v14-latin-200.woff';
source-sans-pro-v14-latin-200italic.woff2';
source-sans-pro-v14-latin-200.woff';
source-sans-pro-v14-latin-200italic.woff2';
```

# Modify your App.js file and remove auto generated styles from the generated project 

```
import { GlobalStyle } from "cc-ui-components/dist/shared/global";
import { Table } from "cc-ui-components";
import { paginationTableData } from "cc-ui-components/dist/components/data";
import { Icon } from "cc-ui-components";
import { SideMenu } from "cc-ui-components";

function App() {
  return (
    <div className="App">
      <GlobalStyle/>
      <Icon icon="normal" aria-hidden />
      <Table data={paginationTableData} isSortable />
      <SideMenu />
    </div>
  );
}

export default App;
```



CHANGELOG.md
## [1.3.5](https://github.com/Manuel-Aullo/ui-components-st-design-system/compare/v1.3.4...v1.3.5) (2021-06-24)


### Bug Fixes

* replaced some texts and logo ([978cc50](https://github.com/Manuel-Aullo/ui-components-st-design-system/commit/978cc50faa0aecaaa01534957f334ae0bda717c9))

## [1.3.4](https://github.com/Manuel-Aullo/ui-components-st-design-system/compare/v1.3.3...v1.3.4) (2021-06-23)


### Bug Fixes

* modified readme file ([30b0b8f](https://github.com/Manuel-Aullo/ui-components-st-design-system/commit/30b0b8febb2b20f5ba1c62ee9d7d664e5a9e5fbb))

## [1.3.3](https://github.com/Manuel-Aullo/ui-components-st-design-system/compare/v1.3.2...v1.3.3) (2021-06-23)


### Bug Fixes

* modified readme file ([c20e889](https://github.com/Manuel-Aullo/ui-components-st-design-system/commit/c20e8892f8153703130acb8a958adcf9fe3a840f))

## [1.3.2](https://github.com/Manuel-Aullo/ui-components-st-design-system/compare/v1.3.1...v1.3.2) (2021-06-23)


### Bug Fixes

* edited readme.md file Merge branch 'master' of github.com:Manuel-Aullo/ui-components-st-design-system ([6711282](https://github.com/Manuel-Aullo/ui-components-st-design-system/commit/67112822f13899c5a4b3da229417ff700f4d9255))
* modified readme file ([421c2bd](https://github.com/Manuel-Aullo/ui-components-st-design-system/commit/421c2bd2cfc07812e6d373373f942d8eea681a2c))

## [1.3.1](https://github.com/Manuel-Aullo/ui-components-st-design-system/compare/v1.3.0...v1.3.1) (2021-06-21)


### Bug Fixes

* include index.js in babel configuration ([5ba237c](https://github.com/Manuel-Aullo/ui-components-st-design-system/commit/5ba237c068a7b8f361ca7ff008de3cacde0cfc91))

# [1.3.0](https://github.com/Manuel-Aullo/ui-components-st-design-system/compare/v1.2.2...v1.3.0) (2021-06-21)


### Features

* modified babel config to include only the desired files ([4a012c9](https://github.com/Manuel-Aullo/ui-components-st-design-system/commit/4a012c989874b98795cecdbfcf8dcb3905a26026))

## [1.2.2](https://github.com/Manuel-Aullo/ui-components-st-design-system/compare/v1.2.1...v1.2.2) (2021-06-21)


### Performance Improvements

* included new components, templates and latest changes, removed webpack config. ([725e877](https://github.com/Manuel-Aullo/ui-components-st-design-system/commit/725e87797d131fbc26fa7b42170c9c34c1816083))

## [1.2.1](https://github.com/Manuel-Aullo/ui-components-st-design-system/compare/v1.2.0...v1.2.1) (2021-04-29)


### Bug Fixes

* changed image loading ([5b88725](https://github.com/Manuel-Aullo/ui-components-st-design-system/commit/5b8872513e50c5f717896097510be2029dbf4d97))

# [1.2.0](https://github.com/Manuel-Aullo/ui-components-st-design-system/compare/v1.1.6...v1.2.0) (2021-04-29)


### Features

* removed import errors ([ebce7a4](https://github.com/Manuel-Aullo/ui-components-st-design-system/commit/ebce7a422ebc58de8b32bbe431f3e74563ff317b))

## [1.1.6](https://github.com/Manuel-Aullo/ui-components-st-design-system/compare/v1.1.5...v1.1.6) (2021-04-28)


### Bug Fixes

* changed fonts and try to fix logo error ([aabd80d](https://github.com/Manuel-Aullo/ui-components-st-design-system/commit/aabd80df650154fc178896fd61a53e1c8d125d0b))

## [1.1.5](https://github.com/Manuel-Aullo/ui-components-st-design-system/compare/v1.1.4...v1.1.5) (2021-04-28)


### Bug Fixes

* changed babel configuration to copy all content from ./src ([dc658d4](https://github.com/Manuel-Aullo/ui-components-st-design-system/commit/dc658d45963c09e7f8fa624f4e438c77c0ce73ff))

## [1.1.4](https://github.com/Manuel-Aullo/ui-components-st-design-system/compare/v1.1.3...v1.1.4) (2021-04-28)


### Bug Fixes

* changed webpack configuration loading fonts and images separated ([245efc4](https://github.com/Manuel-Aullo/ui-components-st-design-system/commit/245efc442a6dcde08164ce4c1582baf8bfd2b88c))

## [1.1.3](https://github.com/Manuel-Aullo/ui-components-st-design-system/compare/v1.1.2...v1.1.3) (2021-04-28)


### Bug Fixes

* changed webpack configuration ([359e3d4](https://github.com/Manuel-Aullo/ui-components-st-design-system/commit/359e3d465ec422bc466027e7c535414732af3306))

## [1.1.2](https://github.com/Manuel-Aullo/ui-components-st-design-system/compare/v1.1.1...v1.1.2) (2021-04-28)


### Bug Fixes

* changed package name ([32857a7](https://github.com/Manuel-Aullo/ui-components-st-design-system/commit/32857a7d8f87807e501ba097b23001c320bfcd2c))

## [1.1.1](https://github.com/Manuel-Aullo/ui-components-st-design-system/compare/v1.1.0...v1.1.1) (2021-04-28)


### Bug Fixes

* changed package name ([4dc2abd](https://github.com/Manuel-Aullo/ui-components-st-design-system/commit/4dc2abdccd9083280c8fab7d8cbf12f10d05737c))
* merging versions changes ([8556943](https://github.com/Manuel-Aullo/ui-components-st-design-system/commit/855694332962dcb543bcfe32685e8cfea0c7769c))

# [1.1.0](https://github.com/Manuel-Aullo/ui-components-st-design-system/compare/v1.0.0...v1.1.0) (2021-04-28)


### Features

* modified data.js ([9a6e682](https://github.com/Manuel-Aullo/ui-components-st-design-system/commit/9a6e682ca190358b2c13d7da8a3afff84ec2d240))

# 1.0.0 (2021-04-28)


### Features

* added commit analyzer to package.json ([17bcbe2](https://github.com/Manuel-Aullo/ui-components-st-design-system/commit/17bcbe288ae3b5683e9d0f77a7da79507dfe5e80))


src/shared/styles.js
import { css } from 'styled-components';
import {SourceSansProWoffLatinItalic} from'./fonts/source-sans-pro-v14-latin-italic.woff';
import {SourceSansProWoffLatinRegular} from'./fonts/source-sans-pro-v14-latin-regular.woff';
import {SourceSansProWoff200} from './fonts/source-sans-pro-v14-latin-200.woff';
import {SourceSansProWoff200Italic} from './fonts/source-sans-pro-v14-latin-200italic.woff';
import {SourceSansProWoff300} from './fonts/source-sans-pro-v14-latin-200.woff';
import {SourceSansProWoff300Italic} from './fonts/source-sans-pro-v14-latin-200italic.woff';
import {SourceSansProWoff600} from './fonts/source-sans-pro-v14-latin-200.woff';
import {SourceSansProWoff600Italic} from './fonts/source-sans-pro-v14-latin-200italic.woff';
import {SourceSansProWoff700} from './fonts/source-sans-pro-v14-latin-200.woff';
import {SourceSansProWoff700Italic} from './fonts/source-sans-pro-v14-latin-200italic.woff';
import {SourceSansProWoff900} from './fonts/source-sans-pro-v14-latin-200.woff';
import {SourceSansProWoff900Italic} from './fonts/source-sans-pro-v14-latin-200italic.woff';
import {SourceSansProWoff2LatinItalic} from'./fonts/source-sans-pro-v14-latin-italic.woff2';
import {SourceSansProWoff2LatinRegular} from'./fonts/source-sans-pro-v14-latin-regular.woff2';
import {SourceSansProWoff2200} from './fonts/source-sans-pro-v14-latin-200.woff2';
import {SourceSansProWoff2200Italic} from './fonts/source-sans-pro-v14-latin-200italic.woff2';
import {SourceSansProWoff2300} from './fonts/source-sans-pro-v14-latin-200.woff';
import {SourceSansProWoff2300Italic} from './fonts/source-sans-pro-v14-latin-200italic.woff2';
import {SourceSansProWoff2600} from './fonts/source-sans-pro-v14-latin-200.woff';
import {SourceSansProWoff2600Italic} from './fonts/source-sans-pro-v14-latin-200italic.woff2';
import {SourceSansProWoff2700} from './fonts/source-sans-pro-v14-latin-200.woff';
import {SourceSansProWoff2700Italic} from './fonts/source-sans-pro-v14-latin-200italic.woff2';
import {SourceSansProWoff2900} from './fonts/source-sans-pro-v14-latin-200.woff';
import {SourceSansProWoff2900Italic} from './fonts/source-sans-pro-v14-latin-200italic.woff2';
// Global style variables
export const background = {
  app: '#F6F9FC',
  appInverse: '#7A8997',
  positive: '#E1FFD4',
  negative: '#FEDED2',
  warning: '#FFF5CF',
};

// Cybercom Specific Colors
// Use these colors when you start to build our own component
export const colors = {
  primary: {
    CCPurple: "#713188",
    CCBlue:"#015599"
  },
  secondary: {
    CCDarkPurple:"#3f194d",
    CCLightPurple: "#eee6f1",
    CCDarkBlue:"#0a3355",
    CCLightBlue:"#e1ebf3"
  },
  neutral: {
    Gray1:"#333333",
    Gray2:"#4f4f4f",
    Gray3:"#828282",
    Gray4:"#bdbdbd",
    Gray5:"#e0e0e0",
    Gray6:"#f2f2f2",
    Gray7:"#f3f7f9",
    CCWhite:"#FFFFFF"
  },
  semantic: {
    CCRed: "#cc142f",
    CCRedLight: "#feebee"
  }

}

// These are colors which are used in example component,
// So, when you start creating component, use the colors above
// Will delete these colors in the future
export const color = {
  // Palette
  primary: '#FF4785', // coral
  secondary: '#1EA7FD', // ocean
  tertiary: '#DDDDDD',

  orange: '#FC521F',
  gold: '#FFAE00',
  green: '#66BF3C',
  seafoam: '#37D5D3',
  purple: '#6F2CAC',
  ultraviolet: '#2A0481',

  // Monochrome
  lightest: '#FFFFFF',
  lighter: '#F8F8F8',
  light: '#F3F3F3',
  mediumlight: '#EEEEEE',
  medium: '#DDDDDD',
  mediumdark: '#999999',
  dark: '#666666',
  darker: '#444444',
  darkest: '#333333',

  border: 'rgba(0,0,0,.1)',

  // Status
  positive: '#66BF3C',
  negative: '#FF4400',
  warning: '#E69D00',
};

export const spacing = {
  padding: {
    small: 10,
    medium: 20,
    large: 30,
  },
  borderRadius: {
    small: 5,
    default: 10,
  },
};

export const font = css`
/* source-sans-pro-200 - latin */
@font-face {
  font-family: 'Source Sans Pro';
  font-style: normal;
  font-weight: 200;
  src: local(''),
       url(${SourceSansProWoff2200}) format('woff2'), /* Chrome 26+, Opera 23+, Firefox 39+ */
       url(${SourceSansProWoff200}) format('woff'); /* Chrome 6+, Firefox 3.6+, IE 9+, Safari 5.1+ */
}
/* source-sans-pro-200italic - latin */
@font-face {
  font-family: 'Source Sans Pro';
  font-style: italic;
  font-weight: 200;
  src: local(''),
       url(${SourceSansProWoff2200Italic}) format('woff2'), /* Chrome 26+, Opera 23+, Firefox 39+ */
       url(${SourceSansProWoff200Italic}) format('woff'); /* Chrome 6+, Firefox 3.6+, IE 9+, Safari 5.1+ */
}
/* source-sans-pro-300 - latin */
@font-face {
  font-family: 'Source Sans Pro';
  font-style: normal;
  font-weight: 300;
  src: local(''),
       url(${SourceSansProWoff2300}) format('woff2'), /* Chrome 26+, Opera 23+, Firefox 39+ */
       url(${SourceSansProWoff300})) format('woff'); /* Chrome 6+, Firefox 3.6+, IE 9+, Safari 5.1+ */
}
/* source-sans-pro-regular - latin */
@font-face {
  font-family: 'Source Sans Pro';
  font-style: normal;
  font-weight: 400;
  src: local(''),
       url(${SourceSansProWoff2LatinRegular}) format('woff2'), /* Chrome 26+, Opera 23+, Firefox 39+ */
       url(${SourceSansProWoffLatinRegular}) format('woff'); /* Chrome 6+, Firefox 3.6+, IE 9+, Safari 5.1+ */
}
/* source-sans-pro-300italic - latin */
@font-face {
  font-family: 'Source Sans Pro';
  font-style: italic;
  font-weight: 300;
  src: local(''),
       url(${SourceSansProWoff2300Italic}) format('woff2'), /* Chrome 26+, Opera 23+, Firefox 39+ */
       url(${SourceSansProWoff300Italic}) format('woff'); /* Chrome 6+, Firefox 3.6+, IE 9+, Safari 5.1+ */
}
/* source-sans-pro-italic - latin */
@font-face {
  font-family: 'Source Sans Pro';
  font-style: italic;
  font-weight: 400;
  src: local(''),
       url(${SourceSansProWoff2LatinItalic}) format('woff2'), /* Chrome 26+, Opera 23+, Firefox 39+ */
       url(${SourceSansProWoffLatinItalic}) format('woff'); /* Chrome 6+, Firefox 3.6+, IE 9+, Safari 5.1+ */
}
/* source-sans-pro-600 - latin */
@font-face {
  font-family: 'Source Sans Pro';
  font-style: normal;
  font-weight: 600;
  src: local(''),
       url(${SourceSansProWoff2600}) format('woff2'), /* Chrome 26+, Opera 23+, Firefox 39+ */
       url(${SourceSansProWoff600}) format('woff'); /* Chrome 6+, Firefox 3.6+, IE 9+, Safari 5.1+ */
}
/* source-sans-pro-600italic - latin */
@font-face {
  font-family: 'Source Sans Pro';
  font-style: italic;
  font-weight: 600;
  src: local(''),
       url(${SourceSansProWoff2600Italic}) format('woff2'), /* Chrome 26+, Opera 23+, Firefox 39+ */
       url(${SourceSansProWoff600Italic}) format('woff'); /* Chrome 6+, Firefox 3.6+, IE 9+, Safari 5.1+ */
}
/* source-sans-pro-700 - latin */
@font-face {
  font-family: 'Source Sans Pro';
  font-style: normal;
  font-weight: 700;
  src: local(''),
       url(${SourceSansProWoff2700}) format('woff2'), /* Chrome 26+, Opera 23+, Firefox 39+ */
       url(${SourceSansProWoff700}) format('woff'); /* Chrome 6+, Firefox 3.6+, IE 9+, Safari 5.1+ */
}
/* source-sans-pro-700italic - latin */
@font-face {
  font-family: 'Source Sans Pro';
  font-style: italic;
  font-weight: 700;
  src: local(''),
       url(${SourceSansProWoff2700Italic}) format('woff2'), /* Chrome 26+, Opera 23+, Firefox 39+ */
       url(${SourceSansProWoff700Italic}) format('woff'); /* Chrome 6+, Firefox 3.6+, IE 9+, Safari 5.1+ */
}
/* source-sans-pro-900 - latin */
@font-face {
  font-family: 'Source Sans Pro';
  font-style: normal;
  font-weight: 900;
  src: local(''),
       url(${SourceSansProWoff2900}) format('woff2'), /* Chrome 26+, Opera 23+, Firefox 39+ */
       url(${SourceSansProWoff900}) format('woff'); /* Chrome 6+, Firefox 3.6+, IE 9+, Safari 5.1+ */
}
/* source-sans-pro-900italic - latin */
@font-face {
  font-family: 'Source Sans Pro';
  font-style: italic;
  font-weight: 900;
  src: local(''),
       url(${SourceSansProWoff2900Italic}) format('woff2'), /* Chrome 26+, Opera 23+, Firefox 39+ */
       url(${SourceSansProWoff900Italic}) format('woff'); /* Chrome 6+, Firefox 3.6+, IE 9+, Safari 5.1+ */
}
`;

export const typography = {
  type: {
    primary: 'Source Sans Pro',
    code: '"SFMono-Regular", Consolas, "Liberation Mono", Menlo, Courier, monospace',
  },
  weight: {
    regular: '400',
    semibold: '600',
    bold: '700',
    black: '900',
  },

  size: {
    s1: '12px',
    s2: '14px',
    s3: '16px',
    m1: '20px',
    m2: '24px',
    m3: '28px',
    l1: '36px',
    l2: '40px',
    l3: '48px',
    code: '90',
  },
};

export const breakpoint = 600;
export const pageMargin = '5.55555';

export const pageMargins = css`
  padding: 0 ${spacing.padding.medium}px;
  @media (min-width: ${breakpoint * 1}px) {
    margin: 0 ${pageMargin * 1}%;
  }
  @media (min-width: ${breakpoint * 2}px) {
    margin: 0 ${pageMargin * 2}%;
  }
  @media (min-width: ${breakpoint * 3}px) {
    margin: 0 ${pageMargin * 3}%;
  }
  @media (min-width: ${breakpoint * 4}px) {
    margin: 0 ${pageMargin * 4}%;
  }
`;

const size= {
  mobileS: '320px',
  mobileM: '375px',
  mobileL: '425px',
  tablet: '768px',
  laptop: '1024px',
  laptopL: '1440px',
  desktop: '2560px'
}

export const device = {
  mobileS: `(min-width: ${size.mobileS})`,
  mobileM: `(min-width: ${size.mobileM})`,
  mobileL: `(min-width: ${size.mobileL})`,
  tablet: `(min-width: ${size.tablet})`,
  laptop: `(min-width: ${size.laptop})`,
  laptopL: `(min-width: ${size.laptopL})`,
  desktop: `(min-width: ${size.desktop})`,
  desktopL: `(min-width: ${size.desktop})`
}
